src/OpenLoco/Ui/ViewportInteraction.cpp

```diff
@@ -401,6 +407,467 @@ namespace OpenLoco::Ui::ViewportInteraction
+        {
+            return false;
+        }
+
+        if (elStation->owner() != CompanyManager::getControllingId())
+        {
+            return false;
+        }
+
+        auto* station = StationManager::get(elStation->stationId());
+        auto args = FormatArguments::mapToolTip();
+        if (Windows::Construction::isStationTabOpen())
+        {
+            args.push(StringIds::stringid_right_click_to_remove);
+        }
+        else
+        {
+            args.push(StringIds::stringid_right_click_to_modify);
+        }
+        args.push(StringIds::quote_string_quote);
+        args.push(station->name);
+        args.push(station->town);
+        return true;
+    }
+
+    // 0x004CDCD9
+    static bool rightOverDock(InteractionArg& interaction)
+    {
+        auto* tileElement = reinterpret_cast<Map::TileElement*>(interaction.object);
+        auto* elStation = tileElement->asStation();
+        if (elStation == nullptr)
+        {
+            return false;
+        }
+
+        if (elStation->isGhost())
+        {
+            return false;
+        }
+
+        if (elStation->owner() != CompanyManager::getControllingId())
+        {
+            return false;
+        }
+
+        auto* station = StationManager::get(elStation->stationId());
+        auto args = FormatArguments::mapToolTip();
+        if (Windows::Construction::isStationTabOpen())
+        {
+            args.push(StringIds::stringid_right_click_to_remove);
+        }
+        else
+        {
+            args.push(StringIds::stringid_right_click_to_modify);
+        }
+        args.push(StringIds::quote_string_quote2);
+        args.push(station->name);
+        args.push(station->town);
+        return true;
+    }
+
+    // 0x004CDE58
+    static bool rightOverTree(InteractionArg& interaction)
+    {
+        auto* tileElement = reinterpret_cast<Map::TileElement*>(interaction.object);
+        auto* tree = tileElement->asTree();
+        if (tree == nullptr)
+        {
+            return false;
+        }
+
+        auto* treeObj = ObjectManager::get<TreeObject>(tree->treeObjectId());
+        FormatArguments::mapToolTip(StringIds::stringid_right_click_to_remove, treeObj->name);
+        return true;
+    }
+
+    // 0x004CE0D8
+    static bool rightOverWall(InteractionArg& interaction)
+    {
+        auto* tileElement = reinterpret_cast<Map::TileElement*>(interaction.object);
+        auto* wall = tileElement->asWall();
+        if (wall == nullptr)
+        {
+            return false;
+        }
+
+        if (!isEditorMode())
+        {
+            return false;
+        }
+        auto* wallObj = ObjectManager::get<WallObject>(wall->wallObjectId());
+        FormatArguments::mapToolTip(StringIds::stringid_right_click_to_remove, wallObj->name);
+        return true;
+    }
+
+    // 0x004CDE78
+    // Note: this is only when in a constructing mode
+    static bool rightOverBuildingConstruct(InteractionArg& interaction)
```

Comment:
On this PR, buildings are right-click removable outside of construction mode. I can't reproduce this on master (c8228154).